pkg/storage/errors.go

```diff
@@ -96,15 +114,24 @@ func IsUnreachable(err error) bool {
 
 // IsTestFailed returns true if and only if err is a write conflict.
 func IsTestFailed(err error) bool {
-	return isErrCode(err, ErrCodeResourceVersionConflicts)
+	return isErrCode(err, ErrCodeResourceVersionConflicts, ErrCodeInvalidUID)
```

Comment:
Is the intent for us to add UID to the api objects?  
b/c right now that's not a property.